File: app/src/main/java/com/akrapovic/soundkit/community/ui/components/ReceiverNotReadyChecklist.kt
package com.akrapovic.soundkit.community.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.unit.dp

object ReceiverNotReadyChecklist {
    val steps: List<String> = listOf(
        "Vehicle switched on (ignition or accessory) so the Sound Kit receiver has power.",
        "Engine running if your install requires it — some receivers stay in status 04 until ready.",
        "Phone within about 20 metres of the receiver.",
        "Receiver green LED indicates power; check the Akrapovič manual if the remote LED blinks white.",
        "Stay parked — retry Open/Close after a few seconds once the car is ready.",
        "Still stuck? Export diagnostics and email support@appsforgood.net.",
    )
}

@Composable
fun ReceiverNotReadyChecklistContent() {
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        Text(
            text = "The receiver is connected but not accepting valve changes yet. Try these steps:",
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
        ReceiverNotReadyChecklist.steps.forEach { step ->
            Text(
                text = "•  $step",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

File: ios/SoundKitCommunity/Views/Home/ReceiverNotReadyChecklist.swift
import SwiftUI

enum ReceiverNotReadyChecklist {
    static let steps = [
        "Switch on the vehicle (ignition or accessory) so the Sound Kit receiver has power.",
        "Start the engine if your installation requires it; some receivers remain on status 04 until ready.",
        "Move the phone within about 20 metres of the receiver.",
        "Check that the receiver green LED shows power. If the remote LED blinks white, consult the Akrapovič manual.",
        "Stay parked, wait a few seconds, then retry Open or Close.",
        "If it remains unavailable, export Diagnostics and email support@appsforgood.net.",
    ]
}

struct ReceiverNotReadyChecklistView: View {
    @Environment(\.garageTheme) private var theme

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Label("Receiver not ready", systemImage: "exclamationmark.triangle.fill")
                .font(.headline)
                .foregroundStyle(.orange)
            Text("The receiver is connected but is not accepting valve changes yet. Try these steps:")
                .font(.subheadline)
                .foregroundStyle(theme.muted)
            ForEach(Array(ReceiverNotReadyChecklist.steps.enumerated()), id: \.offset) { _, step in
                Text("• \(step)")
                    .font(.footnote)
                    .foregroundStyle(theme.muted)
            }
        }
        .padding(16)
        .background(theme.surface)
        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .accessibilityElement(children: .combine)
    }
}
